src/crypto/store/indexeddb-crypto-store-backend.js

```diff
@@ -426,6 +426,74 @@ export class Backend {
         });
     }
 
+    async storeEndToEndSessionProblem(deviceKey, type, fixed) {
+        const txn = this._db.transaction("session_problems", "readwrite");
+        const objectStore = txn.objectStore("session_problems");
+        objectStore.put({
+            deviceKey,
+            type,
+            fixed,
+            time: Date.now(),
+        });
+        return promiseifyTxn(txn);
+    }
+
+    async getEndToEndSessionProblem(deviceKey, timestamp) {
+        let result;
+        const txn = this._db.transaction("session_problems", "readwrite");
+        const objectStore = txn.objectStore("session_problems");
+        const index = objectStore.index("deviceKey");
+        const req = index.getAll(deviceKey);
+        req.onsuccess = (event) => {
+            const problems = req.result;
+            if (!problems.length) {
+                result = null;
+                return;
+            }
+            problems.sort((a, b) => {
+                return a.time - b.time;
+            });
```

Comment:
fwiw you can have multi-level indices on indexeddb now*, so you would be able to index by timestamp, although not 100% sure if you would be able to get the largest ts lower than the given ts. Hopefully this table won't be big enough that it's a problem in terms of efficiency although if we ever want the index in future then could it be useful to add it now to avoid db changes?

*they might still not be supported on Edge but we already have other tables that use them

Reply:
I did take a look at it, but didn't see an easy way to do the query that I needed.